internal/pkg/cli/project_init.go

```diff
@@ -4,60 +4,166 @@
 package cli
 
 import (
+	"errors"
+	"fmt"
+
 	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store"
 	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store/ssm"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/prompt"
 	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
 	"github.com/spf13/cobra"
 )
 
 // InitProjectOpts contains the fields to collect for creating a project.
 type InitProjectOpts struct {
 	ProjectName string
-	manager     archer.ProjectCreator
-	ws          archer.Workspace
+
+	projectStore archer.ProjectStore
+	ws           archer.Workspace
+	prompt       prompter
+}
+
+// NewInitProjectOpts returns a new InitProjectOpts.
+func NewInitProjectOpts() (*InitProjectOpts, error) {
+	ssmStore, err := ssm.NewStore()
+	if err != nil {
+		return nil, err
+	}
+
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, err
+	}
+	return &InitProjectOpts{
+		projectStore: ssmStore,
+		ws:           ws,
+		prompt:       prompt.New(),
+	}, nil
+}
+
+// Ask prompts the user for any required arguments that they didn't provide.
+func (opts *InitProjectOpts) Ask() error {
+	if opts.ProjectName != "" {
+		// User passed in a name as an argument.
+		return nil
+	}
+
+	// If there's a local project, we'll use that and just skip the project question.
+	// Otherwise, we'll load a list of existing projects that the customer can select from.
+	if summary, err := opts.ws.Summary(); err == nil {
+		log.Infof("Looks like you are using a workspace that's registered to project %s. We'll use that as your project.\n", color.HighlightUserInput(summary.ProjectName))
+		opts.ProjectName = summary.ProjectName
+		return nil
+	}
+
+	existingProjects, _ := opts.projectStore.ListProjects()
+	if len(existingProjects) == 0 {
+		log.Infoln("Looks like you don't have any existing projects. Let's create one!")
+		return opts.askNewProjectName()
+	}
+
+	log.Infoln("Looks like you have some projects already.")
+	useExistingProject, err := opts.prompt.Confirm("Would you like to use one of your existing projects?", "", prompt.WithTrueDefault())
+	if err != nil {
+		return fmt.Errorf("new project confirmation: %w", err)
```

Comment:
nit: Is this error message accurate?

Reply:
Changed to "prompt to confirm using existing project"